### PR TITLE
feat(ui5-switch) add new web component

### PR DESCRIPTION
**Pull request by ilhan007**

### CODE REVIEW

File: packages/main/src/Switch.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		super(state);
+	}
+
+	ontap(event) {
+		if (this._tapAllowed(event.ui5target)) {
+			this.toggle();
+		}
+	}
+
+	onkeydown(event) {
```
Comment (MapTo0) [inline packages/main/src/Switch.js]: maybe use the `onsap____`? We need to discuss if it is really needed

File: packages/main/src/SwitchTemplateContext.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			classes: { main: mainClasses },
+		};
+
+		return context;
+	}
+
+	static getMainClasses(state) {
+		return {
+			"ui5-switch": true,
+			"ui5-switch--disabled": state.disabled,
```
Comment (MapTo0) [inline packages/main/src/SwitchTemplateContext.js]: classes look good to me, but we need to discuss if we are staying with that BEM

File: packages/main/src/themes/base/Switch.less [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+	text-align: center;
+	color: @sapUiBaseText;
+	white-space: nowrap;
+	user-select: none;
+	-webkit-user-select: none;
+	-ms-user-select: none;
+}
+
+.ui5-switch-text--on {
+	left: calc(~"-100% + 1.9125rem");
```
Comment (MapTo0) [inline packages/main/src/themes/base/Switch.less]: hmm, should you escape the whole `calc` expression like `~"calc(100% .. )"`

File: packages/main/src/Switch.js (hunk 1 above)
Comment (MapTo0) [reply packages/main/src/Switch.js]: by `it` I mean `onsap__` psuedo events generation

File: packages/main/src/themes/base/Switch.less (hunk 3 above)
Comment (ilhan007) [reply packages/main/src/themes/base/Switch.less]: > hmm, should you escape the whole `calc` expression like `~"calc(100% .. )"`

Yes, in the less files we should escape the calc, once we move the styles in css, we can remove the escaping.

File: packages/main/src/SwitchTemplateContext.js (hunk 2 above)
Comment (ilhan007) [reply packages/main/src/SwitchTemplateContext.js]: > classes look good to me, but we need to discuss if we are staying with that BEM

For sure we should discuss it, I liked it as well.

**Pull request merged**